osu.Game.Tests/Visual/TestCaseUserProfile.cs

```diff
@@ -95,6 +94,32 @@ private void checkSupporterTag(bool isSupporter)
                 AddAssert("no supporter", () => profile.Header.SupporterTag.Alpha == 0);
         }
 
+        private User createDummyUser(Badge[] badges)
+        {
+            return new User
+            {
+                Username = @"Somebody",
+                Id = 1,
+                Country = new Country { FullName = @"Alien" },
+                CoverUrl = @"https://osu.ppy.sh/images/headers/profile-covers/c1.jpg",
+                JoinDate = DateTimeOffset.Now.AddDays(-1),
+                LastVisit = DateTimeOffset.Now,
+                Age = 1,
+                ProfileOrder = new[] { "me" },
+                Statistics = new UserStatistics
+                {
+                    Ranks = new UserStatistics.UserRanks { Global = 2148, Country = 1 },
+                    PP = 4567.89m,
+                },
+                RankHistory = new User.RankHistoryData
+                {
+                    Mode = @"osu",
+                    Data = Enumerable.Range(2345, 45).Concat(Enumerable.Range(2109, 40)).ToArray()
+                },
+                Badges = badges
+            };
+        }
```

Comment:
Instead of repurposing this testcase for badges, create another testcase with _just_ the `BadgeContainer` in it.

We want to test individual components in general.